### PR TITLE
Cleanup before next pr

### PR DESCRIPTION
**Pull request by PopGoesTheWza**
Fixes no issue

- [x] `npm run test` succeeds.
- [x] `npm run lint` succeeds.
- [ ] Appropriate changes to README are included in PR.

Some clean-up before I attempt a PR to specify dot files from cli options and environment variables.

### CODE REVIEW

File: src/apiutils.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import * as fuzzy from 'fuzzy';
 import { script_v1 } from 'googleapis';
+// setup inquirer
```
Comment (grant) [inline src/apiutils.ts]: Move to `inquirer.registerPrompt`.

File: src/commands/clone.ts [hunk 2]
```diff
@@ -37,10 +21,13 @@ export default async (scriptId: string, versionNumber: number, cmd: { rootDir: s
   scriptId = scriptId ? extractScriptId(scriptId) : await getScriptId();
   spinner.setSpinnerTitle(LOG.CLONING);
   const rootDir = cmd.rootDir;
-  saveProject({
-    scriptId,
-    rootDir,
-  }, false);
+  saveProject(
```
Comment (grant) [inline src/commands/clone.ts]: Can we undo this? Is it required?

File: src/commands/clone.ts [hunk 3]
```diff
@@ -37,10 +21,13 @@ export default async (scriptId: string, versionNumber: number, cmd: { rootDir: s
...
+    {
+      scriptId,
+      rootDir,
+    },
+    false,
```
Comment (grant) [inline src/commands/clone.ts]: I think removing the trailing comma here will make this wrap on one line.

File: src/commands/logs.ts [hunk 4]
```diff
@@ -1,9 +1,10 @@
 import chalk from 'chalk';
-import {logging_v2} from 'googleapis';
-import {
-  loadAPICredentials,
-  logger,
-} from './../auth';
+import { logging_v2 } from 'googleapis';
+// setup inquirer
```
Comment (grant) [inline src/commands/logs.ts]: Ditto. Move down.

File: src/commands/openCmd.ts [hunk 5]
```diff
@@ -1,25 +1,20 @@
 import { script_v1 } from 'googleapis';
-import {
-  loadAPICredentials,
-  script,
-} from './../auth';
+// setup inquirer
```
Comment (grant) [inline src/commands/openCmd.ts]: It looks like we set up inquirer a lot.
Can you move it to a separate file and import it?

File: src/commands/run.ts [hunk 6]
```diff
@@ -109,16 +92,18 @@ async function runFunction(functionName: string, params: string[], scriptId: str
           // The 401 is probably due to this error:
           // "Error: Local client credentials unauthenticated. Check scopes/authorization.""
           // This is probably due to the OAuth client not having authorized scopes.
-          console.log(`` +
-            `Hey! It looks like you aren't authenticated for the scopes required by this script.
+          console.log(
+            `` +
+              `Hey! It looks like you aren't authenticated for the scopes required by this script.
```
Comment (grant) [inline src/commands/run.ts]: Weird spacing.

File: src/dotfile.ts [hunk 7]
```diff
@@ -14,13 +14,16 @@
...
+import * as findParentDir from 'find-parent-dir';
 import { Credentials } from 'google-auth-library';
 import { OAuth2ClientOptions } from 'google-auth-library/build/src/auth/oauth2client';
 
 const dotf = require('dotf');
 const read = require('read-file');
-import * as findParentDir from 'find-parent-dir';
-const splitLines = require('split-lines');
+const splitLines: (
+  str :string,
```
Comment (grant) [inline src/dotfile.ts]: Format.

File: src/utils.ts [hunk 8]
```diff
@@ -8,8 +8,9 @@ import * as pluralize from 'pluralize';
 import { ClaspToken, DOT, DOTFILE, ProjectSettings } from './dotfile';
 import { URL } from './urls';
 
-const ucfirst = require('ucfirst');
-const isOnline = require('is-online');
+// const ucfirst = require('ucfirst');
```
Comment (grant) [inline src/utils.ts]: Remove.

File: tests/functions.ts [hunk 9]
```diff
@@ -1,12 +1,14 @@
...
 import {
-  CLASP_SETTINGS,
   CLASP_PATHS,
+  CLASP_SETTINGS,
   TEST_APPSSCRIPT_JSON_WITHOUT_RUN_API,
   TEST_APPSSCRIPT_JSON_WITH_RUN_API,
 } from './constants';
 
-const copyFileSync = require('fs-copy-file-sync');
+// polyfill fo fs.copyFileSync (Added in: v8.5.0)
```
Comment (grant) [inline tests/functions.ts]: Remove this if it isn't needed.

File: tests/functions.ts [hunk 10]
```diff
@@ -35,13 +37,16 @@ export const rndStr = () => Math.random().toString(36).substr(2);
 export const backupSettings = () => {
   // fs.copyFileSync isn't supported in Node 6/7
```
Comment (grant) [inline tests/functions.ts]: We don't support Node 6/7 anymore since it's unsupported by Node itself. Please remove.

File: tests/functions.ts [hunk 11]
```diff
@@ -35,13 +37,16 @@ export const rndStr = () => Math.random().toString(36).substr(2);
...
   if (fs.existsSync(CLASP_PATHS.rcGlobal)) {
-    copyFileSync(CLASP_PATHS.rcGlobal, `${CLASP_PATHS.rcGlobal}~`);
+    // copyFileSync(CLASP_PATHS.rcGlobal, `${CLASP_PATHS.rcGlobal}~`);
```
Comment (grant) [inline tests/functions.ts]: Please remove.

File: src/commands/clone.ts (hunk 2 above)
Comment (PopGoesTheWza) [reply src/commands/clone.ts]: I can manually reformat as

```ts
  saveProject({
    scriptId,
    rootDir,
  }, false);
```

but next `npm run prettier` will change it again (unless we tweak prettier config maybe)

File: src/commands/clone.ts (hunk 3 above)
Comment (PopGoesTheWza) [reply src/commands/clone.ts]: `npm run prettier` (at least as it is configured now) format it like this

File: src/commands/openCmd.ts (hunk 5 above)
Comment (PopGoesTheWza) [reply src/commands/openCmd.ts]: Sure... but in another PR if you don't mind

File: src/commands/run.ts (hunk 6 above)
Comment (PopGoesTheWza) [reply src/commands/run.ts]: `npm run prettier` (at least as it is configured now) format it like this

Changing how the string is created will change how it goes "prettier" probably

Comment (PopGoesTheWza) [reply src/commands/run.ts]: Changed it slightly and it is now "normal"

File: src/commands/login.ts [hunk 12]
```diff
@@ -49,12 +38,13 @@ export default async (options: { localhost?: boolean; creds?: string }) => {
...
     console.log(`Authorizing with the following scopes:`);
-    oauthScopes.forEach((scope) => {
+    oauthScopes.forEach(scope => {
       console.log(scope);
     });
     console.log('');
-    console.log(`NOTE: The full list of scopes your project may need` +
-      ` can be found at script.google.com under:`);
+    console.log(
+      `NOTE: The full list of scopes your project may need` + ` can be found at script.google.com under:`,
```
Comment (grant) [inline src/commands/login.ts]: This line is too long. Unneeded plus.

File: src/commands/list.ts [hunk 13]
```diff
@@ -1,18 +1,13 @@
...
-  LOG,
-  checkIfOnline,
-  logError,
-  spinner,
-} from '../utils';
+import { ERROR, LOG, checkIfOnline, logError, spinner } from '../utils';
 
-const ellipsize = require('ellipsize');
+const ellipsize: (
+  str: string | undefined,
```
Comment (grant) [inline src/commands/list.ts]: str?: string,

File: src/commands/openCmd.ts [hunk 14]
```diff
@@ -1,25 +1,20 @@
...
-  getProjectSettings,
-  getWebApplicationURL,
-  logError,
-} from './../utils';
+import { ERROR, LOG, checkIfOnline, getProjectSettings, getWebApplicationURL, logError } from './../utils';
 
-const ellipsize = require('ellipsize');
-import * as open from 'open';
+const ellipsize: (
+  str: string | undefined,
```
Comment (grant) [inline src/commands/openCmd.ts]: str?: string,

File: src/utils.ts [hunk 15]
```diff
@@ -320,13 +319,10 @@ export function getAPIFileType(filePath: string): string {
 export async function checkIfOnline() {
   // If using a proxy, return true since `isOnline` doesn't work.
   // @see https://github.com/googleapis/google-api-nodejs-client#using-a-proxy
-  if (process.env.HTTP_PROXY || process.env.HTTPS_PROXY) {
+  if (process.env.HTTP_PROXY || process.env.HTTPS_PROXY || (await isOnline())) {
     return true;
   }
-  if (!(await isOnline())) {
-    logError(null, ERROR.OFFLINE);
-    process.exit(1);
```
Comment (grant) [inline src/utils.ts]: I think `process.exit(1);` is needed.

Comment (PopGoesTheWza) [reply src/utils.ts]: No. `logError()` never returns and end with its own `process.exit(1)`

File: src/commands/list.ts (hunk 13 above)
Comment (PopGoesTheWza) [reply src/commands/list.ts]: I double check this package code. If str is not a string, an empty string is used. This allows to call the function with null and undefined.

**Comment by grant** [comment]
Thanks!

**Comment by PopGoesTheWza** [comment]
Welcome @grant 

Who implemented the `inquirer-autocomplete-prompt`? The code here doesn't make sense with its documentation.

**Comment by grant** [comment]
Not sure what you're referring to. Use git blame to find out who implemented it here.

**Comment by PopGoesTheWza** [comment]
#394 is the PR where package `inquirer-autocomplete-prompt` was introduced. Currently it does not seems it is really used (it is imported, it is registered, but never with a used `prompt`). I'll drop it in next PR.

@takanakahiko since the above PR is yours, can you give your opinion?

**Comment by takanakahiko** [comment]
Currently `autocomplete ` use in  https://github.com/google/clasp/blob/267209d3d3a848c6e83f140cfa216d1e80a1a65d/src/apiutils.ts#L41

**Comment by PopGoesTheWza** [comment]
@takanakahiko thanks

I am almost done in refactoring all `inquirer` uses into a single inquirer.ts file

**Comment by grant** [comment]
Note, if you want to move the autocomplete logic to it's own file, that would be great!

**Comment by PopGoesTheWza** [comment]
@grant I moved as much as possible inside inquirer.ts (cf. Open PR)

**Pull request merged**